File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/LevelTipDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.LevelTip;

import java.util.List;

@Dao
public interface LevelTipDao {

    @Query("SELECT * FROM level_tip ORDER BY id")
    LiveData<List<LevelTip>> getLevelTips();

    @Query("SELECT * FROM level_tip WHERE id = :levelTipId")
    LiveData<LevelTip> getLevelTipById(int levelTipId);

    @Insert
    void insertLevelTip(LevelTip levelTip);

    @Insert
    void insertLevelTips(List<LevelTip> levelTips);

    @Query("DELETE FROM level_tip")
    void deleteAll();

    @Delete
    void deleteLevelTip(LevelTip levelTip);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateLevelTip(LevelTip levelTip);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateLevelTips(List<LevelTip> levelTips);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/LevelTipDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.LevelTip

@Dao
interface LevelTipDao {
    @Query("SELECT * FROM level_tip ORDER BY id")
    fun getLevelTips(): LiveData<List<LevelTip>>

    @Query("SELECT * FROM level_tip WHERE id = :levelTipId")
    fun getLevelTipById(levelTipId: Int): LiveData<LevelTip>

    @Insert
    fun insertLevelTip(levelTip: LevelTip)

    @Insert
    fun insertLevelTips(levelTips: List<LevelTip>)

    @Query("DELETE FROM level_tip")
    fun deleteAll()

    @Delete
    fun deleteLevelTip(levelTip: LevelTip)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateLevelTip(levelTip: LevelTip)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateLevelTips(levelTips: List<LevelTip>)
}